### PR TITLE
Gemm fp16 test

### PR DESCRIPTION
**Pull request by rocking5566**
Add gemm fp16 test

### CODE REVIEW

File: test/gemm/gemm_fp16.cpp [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+
+    using RowMajor    = ck::tensor_layout::gemm::RowMajor;
+    using ColumnMajor = ck::tensor_layout::gemm::ColumnMajor;
+
+    bool res = true;
+    std::vector<DeviceGemmPtr_> gemmPtrs;
+
+    ck::tensor_operation::device::device_gemm_instance::
+        add_device_gemm_xdl_c_shuffle_f16_f16_f16_km_kn_mn_instances(gemmPtrs);
+
```
Comment (ltqin) [inline test/gemm/gemm_fp16.cpp:54]: Please add a check to check whether the queue is empty

File: test/gemm/gemm_fp16.cpp [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+                                       RowMajor,
+                                       ColumnMajor,
+                                       RowMajor,
+                                       PassThrough,
+                                       PassThrough,
+                                       PassThrough>{}(gemmPtr);
+    }
+
+    std::cout << "TestGemm ..... " << (res ? "SUCCESS" : "FAILURE") << std::endl;
+}
```
Comment (ltqin) [inline test/gemm/gemm_fp16.cpp:124]: it is better to write "return 0 "

File: test/gemm/gemm_fp32.cpp [hunk 3]
```diff
@@ -32,106 +32,96 @@ namespace ck {
...
+                                       CDataType,
+                                       RowMajor,
+                                       ColumnMajor,
+                                       RowMajor,
+                                       PassThrough,
+                                       PassThrough,
+                                       PassThrough>{}(gemmPtr);
     }
 
     std::cout << "TestGemm ..... " << (res ? "SUCCESS" : "FAILURE") << std::endl;
```
Comment (ltqin) [inline test/gemm/gemm_fp32.cpp:127]: same with above

File: test/include/test_util.hpp [hunk 4]
```diff
@@ -54,6 +54,49 @@ check_err(const std::vector<T>& out,
     return res;
 }
 
+bool check_err(const std::vector<_Float16>& out,
+                   const std::vector<_Float16>& ref,
+                   const std::string& msg,
+                   _Float16 rtol = static_cast<_Float16>(1e-3f),
+                   _Float16 atol = static_cast<_Float16>(1e-3f))
+{
```
Comment (ltqin) [inline test/include/test_util.hpp:62]: change _Float16 to ck::half_t?

**Comment by asroy** [comment]
Please use ckProfiler to dp testing, instead of writting new test  code

You can reuse this function:
https://github.com/ROCmSoftwarePlatform/composable_kernel/blob/925ca9fcd61ce4cf6d6e7c6b76dad1500ad4f543/profiler/include/profile_gemm_impl.hpp#L72

**Review by asroy — CHANGES_REQUESTED** [review]
Please use ckProfiler to do testing, instead of writting new test  code

You can reuse this function:
https://github.com/ROCmSoftwarePlatform/composable_kernel/blob/925ca9fcd61ce4cf6d6e7c6b76dad1500ad4f543/profiler/include/profile_gemm_impl.hpp#L72

**Pull request closed without merging**